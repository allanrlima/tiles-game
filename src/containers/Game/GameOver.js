import React from "react";
import styled from "styled-components";
import { Input } from "../../components/Input/index";

const Wrapper = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  z-index: 10;
  background-color: #000;
  width: 100%;
  height: 100%;
  font-size: 32px;
  color: #fff;
  padding: 16px;
`;

const Title = styled.div`
  text-transform: uppercase;
  display: flex;
  justify-content: center;
  text-indent: 0.7rem;
  font-weight: 300;
  white-space: nowrap;
  letter-spacing: 0.7rem;
`;

export const GameOver = ({ onSubmit, onChangeName }) => (
  <Wrapper>
    <Title>Game over ):</Title>
    <form onSubmit={onSubmit}>
      <Input
        label="Insert your name and press enter, to register your record"
        onChange={onChangeName}
      />
    </form>
  </Wrapper>
);
