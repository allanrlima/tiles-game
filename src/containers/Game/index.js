import React, { Component } from "react";
import { Container } from "../../components/Container/index";
import styled from "styled-components";
import { TilesWrapper } from "../../components/TilesWrapper/index";

const Title = styled.div`
  text-transform: uppercase;
  display: flex;
  justify-content: center;
`;

const Step = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 8px;
`;

class Game extends Component {
  state = {
    step: 1
  };

  getTilesNumberByStep = () => {
    const { step } = this.state;
    const tilesNumber = Math.pow(step + 1, 2);
    return tilesNumber;
  };

  getTiles = () => {
    const tilesNumber = this.getTilesNumberByStep();

    const tile = {
      color: "red",
      unique: false
    };

    const tiles = Array(tilesNumber).fill(tile);
    return tiles;
  };

  render() {
    const { step } = this.state;
    return (
      <Container>
        <Title>Tiles Game</Title>
        <Step>Step: {step}</Step>
        <TilesWrapper tiles={this.getTiles()} />
      </Container>
    );
  }
}

export default Game;
