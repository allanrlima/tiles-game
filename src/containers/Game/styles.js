import styled from "styled-components";

export const Title = styled.div`
  text-transform: uppercase;
  text-indent: 0.7rem;
  font-weight: 300;
  white-space: nowrap;
  letter-spacing: 0.7rem;
`;

export const Step = styled.div`
  font-weight: 600;
`;

export const HallOfFameButton = styled.div`
  width: 160px;
`;
