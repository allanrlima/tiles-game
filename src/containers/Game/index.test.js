import React from "react";
import { shallow } from "enzyme";

import Game from ".";

describe("Game", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Game />);
    expect(wrapper).toMatchSnapshot();
  });
  describe("getTilesNumberByStep()", () => {
    it("should return the number of tiles using the current step", () => {
      const wrapper = shallow(<Game />);
      const steps = [1, 2, 3, 4, 5, 6];
      const stepsResut = [4, 9, 16, 25, 36, 49];
      steps.forEach((step, index) => {
        wrapper.setState({ step });
        const tilesNumber = wrapper.instance().getTilesNumberByStep();
        expect(tilesNumber).toBe(stepsResut[index]);
      });
    });
  });
  describe("getTiles()", () => {
    it("Should return an array of tiles using the current step", () => {
      const wrapper = shallow(<Game />);
      const tiles = wrapper.instance().getTiles();
      expect(tiles).toEqual([
        { color: "red", unique: false },
        { color: "red", unique: false },
        { color: "red", unique: false },
        { color: "red", unique: false }
      ]);
    });
  });
});
