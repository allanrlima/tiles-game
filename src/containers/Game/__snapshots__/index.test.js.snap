// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Game />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Tiles Game
        </ForwardRef>,
        <ForwardRef>
          Step: 
          1
        </ForwardRef>,
        <TilesWrapper
          tiles={
            Array [
              Object {
                "color": "red",
                "unique": false,
              },
              Object {
                "color": "red",
                "unique": false,
              },
              Object {
                "color": "red",
                "unique": false,
              },
              Object {
                "color": "red",
                "unique": false,
              },
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Tiles Game",
        },
        "ref": null,
        "rendered": "Tiles Game",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  text-transform: uppercase;
  display: flex;
  justify-content: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Step: ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Step: ",
          1,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  justify-content: center;
  margin-top: 8px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "tiles": Array [
            Object {
              "color": "red",
              "unique": false,
            },
            Object {
              "color": "red",
              "unique": false,
            },
            Object {
              "color": "red",
              "unique": false,
            },
            Object {
              "color": "red",
              "unique": false,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Tiles Game
          </ForwardRef>,
          <ForwardRef>
            Step: 
            1
          </ForwardRef>,
          <TilesWrapper
            tiles={
              Array [
                Object {
                  "color": "red",
                  "unique": false,
                },
                Object {
                  "color": "red",
                  "unique": false,
                },
                Object {
                  "color": "red",
                  "unique": false,
                },
                Object {
                  "color": "red",
                  "unique": false,
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Tiles Game",
          },
          "ref": null,
          "rendered": "Tiles Game",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  text-transform: uppercase;
  display: flex;
  justify-content: center;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Step: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Step: ",
            1,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  justify-content: center;
  margin-top: 8px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "tiles": Array [
              Object {
                "color": "red",
                "unique": false,
              },
              Object {
                "color": "red",
                "unique": false,
              },
              Object {
                "color": "red",
                "unique": false,
              },
              Object {
                "color": "red",
                "unique": false,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
